.travis.yml

```diff
@@ -5,20 +5,21 @@ python:
   - "2.7"
   - "3.4"
   - "3.6"
+  - "3.7"
 before_install:
   - sudo apt-get install liblapack-dev
   - pip install --upgrade pip pytest
-  - pip install wheel cython numpy scipy codecov pytest-cov
-  - if $TRAVIS_PYTHON_VERSION == "3.6"; then
-        pip install scikit-learn;
-    else
-        pip install scikit-learn==0.20.3;
-    fi
+  - pip install wheel cython numpy scipy codecov pytest-cov scikit-learn flake8
   - if [[ ($TRAVIS_PYTHON_VERSION == "3.6") ||
-          ($TRAVIS_PYTHON_VERSION == "2.7")]]; then
+          ($TRAVIS_PYTHON_VERSION == "3.7") ||
+          ($TRAVIS_PYTHON_VERSION == "2.7") ]]; then
               pip install git+https://github.com/skggm/skggm.git@a0ed406586c4364ea3297a658f415e13b5cbdaf8;
     fi
 script:
+  # we run flake8 to ensure syntax (no need to run it on every version)
+  - if [[ $TRAVIS_PYTHON_VERSION == "3.7" ]];
+      then flake8 --ignore=E111,E114,W504;
```

Comment:
I think we should also ignore the other errors that are ignored by default by flake8, just like sklearn does, to avoid unimportant failures which could pop up in subsequent PRs

Reply:
I agree, I used the syntax `--extend-ignore` (https://flake8.pycqa.org/en/3.7.8/user/options.html?highlight=extend-ignore#cmdoption-flake8-extend-ignore), so that flake8 will ignore the default errors (including W504), and also E111 and E114